packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -574,13 +586,15 @@ export function TransferPanel() {
 
       const destinationChainId = latestNetworks.current.destinationChain.id
 
-      const sourceChainErc20Address = isDepositMode
-        ? selectedToken?.address
-        : selectedToken?.l2Address
+      const sourceChainErc20Address =
+        transferMode === 'deposit'
```

Comment:
Shouldn't we also check for teleport `transferMode` here? Same in `destinationChainErc20Address`.